File: src/test/java/utils/TestUtils.java
package utils;

import com.google.common.collect.Lists;
import java.io.IOException;
import java.util.List;
import javax.swing.text.BadLocationException;
import javax.swing.text.Document;
import static junit.framework.TestCase.assertNotNull;
import org.jetbrains.kotlin.utils.ProjectUtils;
import org.openide.filesystems.FileObject;
import org.openide.util.Exceptions;

/**
 *
 * @author Alexander.Baratynski
 */
public class TestUtils {
    
    public static Integer getCaret(Document doc) {
        try {
            return doc.getText(0, doc.getLength()).indexOf("<caret>");
        } catch (BadLocationException ex) {
            Exceptions.printStackTrace(ex);
        }
        
        return null;
    }
    
    public static List<Integer> getCarets(Document doc, int startOffset) {
        List<Integer> carets = Lists.newArrayList();
        
        try {
            int caretOffset = doc.getText(startOffset, doc.getLength()).indexOf("<caret>");
            if (caretOffset >=0) {
                carets.add(caretOffset);
                carets.addAll(getCarets(doc, caretOffset + "<caret>".length()));
            }
        } catch (BadLocationException ex) {
            Exceptions.printStackTrace(ex);
        }
        
        return carets;
    }
    
    public static Document getDocumentForFileObject(FileObject dir, String fileName) {
        FileObject file = dir.getFileObject(fileName);
        
        assertNotNull(file);
        
        return getDocumentForFileObject(file);
    }
    
    public static Document getDocumentForFileObject(FileObject fo) {
        Document doc = null;
        try {
            doc = ProjectUtils.getDocumentFromFileObject(fo);
        } catch (IOException ex) {
            Exceptions.printStackTrace(ex);
        }
        
        assertNotNull(doc);
        
        return doc;
    }

    public static List<FileObject> getAllKtFilesInFolder(FileObject folder) {
        List<FileObject> ktFiles = Lists.newArrayList();
        for (FileObject child : folder.getChildren()) {
            if (child.hasExt("kt")) {
                ktFiles.add(child);
            }
        }
        
        return ktFiles;
    }
    
}

File: src/test/java/utils/testUtils.kt
/*******************************************************************************
 * Copyright 2000-2016 JetBrains s.r.o.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 *******************************************************************************/
package utils

import javax.swing.text.Document
import junit.framework.TestCase.assertNotNull
import org.jetbrains.kotlin.utils.ProjectUtils
import org.openide.filesystems.FileObject

fun getCaret(doc: Document): Int = doc.getText(0,doc.length).indexOf("<caret>")

fun getDocumentForFileObject(fo: FileObject) = ProjectUtils.getDocumentFromFileObject(fo)

fun getDocumentForFileObject(dir: FileObject, fileName: String): Document {
    val file = dir.getFileObject(fileName)
    
    assertNotNull(file)
    
    return getDocumentForFileObject(file)
}

fun getAllKtFilesInFolder(folder: FileObject) = folder.children.filter{ it.hasExt("kt") }